### DISCUSSION TITLE
Awesome work!

### DISCUSSION THREAD

**Discussion by yrashk** [rhaiscript/rhai#336] [General]
Hey,

Just wanted to chime in and say that I really like the objectives of Rhai (rugged, sandboxed, extendable embedded scripting language) and the active release schedule. Special :hearts: goes to the Rhai Book, as it is fairly comprehensive and is *very* well written.

I am already using it in https://github.com/bpxe/bpxe, though it's rudimentary as the project is in its early stages, but it's coming together as I write this (I am working on further integration right now!).

Will be happy to contribute what I can and drive it towards 1.0!

Yurii.

**Comment by yrashk** [comment]
By the way, @schungx -- I would be happy to sponsor rhai.rs domain if you're interested.

**Reply by schungx** [reply]
Wow, thanks!  A `rhai.rs` (or `rhaiscript.rs` to be consistent) would be nice!  We can map it to `rhaiscript.github.io`...

Although we don't have much of a web site right now...

Maybe wait until we have a _real_ website up?

**Reply by yrashk** [reply]
I'd personally prefer rhai.rs for its awesome brevity :) What do you think?

We can publish the book there and then turn it into a real website when the time comes.

Basically, we can just add the CNAME file (directly or through [repository](https://github.com/rhaiscript/book/tree/gh-pages)'s Settings) to point to rhai.rs and voila. I think this will also redirect rhaiscript.github.io to rhai.rs

**Reply by schungx** [reply]
`rhai.rs` is fine with me, and I agree it sounds nice!

However, right now a number of separate sub-sites are published directly from the org directly into `rhaiscript.github.io`, including the playground and the book.

Is it possible to simply redirect `rhai.rs` to `rhaiscript.github.io`?

**Reply by yrashk** [reply]
Well, that's the idea -- whatever is `rhaiscript.github.io`, once you will add a CNAME file with `rhai.rs` (or through Settings) will become `rhai.rs` and GitHub will redirect `rhaiscript.github.io` to it. Does this sound good?

**Reply by schungx** [reply]
Wonder where do I set this CNAME setting...

**Reply by yrashk** [reply]
https://docs.github.com/en/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain

I just got the domain and configured it for GitHub pages :)

**Reply by yrashk** [reply]
Might need to wait a bit before configuring CNAME -- as soon as `rhai.rs` resolves to any of these, should be good to go:

```
185.199.108.153
185.199.109.153
185.199.110.153
185.199.111.153
```

**Reply by schungx** [reply]
OK I already created the CNAME file for https://github.com/rhaiscript/rhaiscript.github.io

**Reply by yrashk** [reply]
Resolves on my end (through Google DNS (8.8.8.8), at least)

**Reply by yrashk** [reply]
I am already able to access the website at rhai.rs. You might want to turn "Enforce HTTPS" setting on as well to avoid browsers showing that rhai.rs is not using a secure connection.

**Reply by schungx** [reply]
Works fine!

**Reply by yrashk** [reply]
(It will take them up to a day to produce a valid certificate but then it's all nice and "green" looking :)

**Reply by schungx** [reply]
Yup, will wait a bit.  Then I'll change all references to `rhaiscript.github.io` to `rhai.rs`!

**Reply by yrashk** [reply]
Did you turn on the Enforce HTTPS setting? https://docs.github.com/en/github/working-with-github-pages/securing-your-github-pages-site-with-https#enforcing-https-for-your-github-pages-site

**Reply by yrashk** [reply]
Seems like the HTTPS certificate is valid now.

**Reply by schungx** [reply]
Done!

**Reply by yrashk** [reply]
Sweet, I can confirm it works!

By the way, what's the best way to ask questions / discuss matters related to Rhai itself? Should it be Discussions here, Issues, or chat somewhere? I've got some questions I am not able to easily answer.

**Reply by schungx** [reply]
I find that it is qiucker to do it in Discord...  But raising an issue here is the best way to keep track of feature request or bug reports.

**Comment by frostu8** [comment]
I second this; Rhai is probably going to be my choice of scripting for any games I pursue in the future. Lua is not a very great scripting language at all, so I've been looking for alternatives like this.

**Reply by schungx** [reply]
Well, LuaJIT is extremely fast and light on the CPU.  You can actually implement you entire game in Lua if you want, and move code to C++ only when it is stable enough.

With Rhai, you'll have to be more selective what you put in scripting.  You'll need to have most of the core infrastructure provided in Rust itself, only the behavioral controls in script.  So there is a difference.

**Reply by frostu8** [reply]
> Well, LuaJIT is extremely fast and light on the CPU. You can actually implement you entire game in Lua if you want, and move code to C++ only when it is stable enough.
> 
> With Rhai, you'll have to be more selective what you put in scripting. You'll need to have most of the core infrastructure provided in Rust itself, only the behavioral controls in script. So there is a difference.

My distaste for Lua isn't necessarily with its speed, it's more a problem with its syntax and composure. I go into making projects with a Little Big Planet philosophy, where I think the engine should be super generic, and all the specifics are built on top of a Rust generic foundation, so Rhai is naturally a better pick for me.

**Reply by schungx** [reply]
Yes, that's what Rhai is designed for - to provide a controls/logic layer over generic Rust functionalities.

**Comment by schungx** [comment]
Closing this for now. Feel free to reopen if necessary.
